src/test/java/org/broadinstitute/hellbender/utils/read/ReadUtilsUnitTest.java

```diff
@@ -616,4 +614,118 @@ public void testNotValidExpectedReadOrder(
             );
     }
 
+    @Test
+    public void testOptionalIntAttributeOnSAMRecord() {
+        final SAMRecord record = ArtificialReadUtils.createArtificialSAMRecord(
```

Comment:
You could (but this is not a required change), convert the SAMRecord to GATKRead or vice versa and remove code duplication